test/common/testlib.py

```diff
@@ -466,9 +466,13 @@ def run(self, result=None):
                 self.currentResult = result
                 super(MachineCase, self).run(result)
             except RetryError, ex:
-                self.doCleanups()
-                sys.stderr.write("{0}\n".format(ex))
-                continue
+                if retry < 5:
```

Comment:
RetryError should only be fired when we do in fact want to retry. The policy dictates that.